modules/unsupported/mongodb/src/main/java/org/geotools/data/mongodb/FilterToMongo.java

```diff
@@ -621,6 +621,8 @@ Object encodeLiteral(Object literal, Class<?> targetType) {
             }
             // by default, return string as is
             return literal;
+        } else if (literal instanceof Number) {
+            return literal;
```

Comment:
Agreed!  I have updated the fix to return the literal if Long, Integer or Double.  For backwards compatibility the else still returns literal.toString().

Thanks for all the help and input, and please let me know if this PR needs anything else.